dotnet/src/SemanticKernel/Configuration/KernelConfig.cs

```diff
@@ -15,9 +18,76 @@ namespace Microsoft.SemanticKernel.Configuration;
 public sealed class KernelConfig
 {
     /// <summary>
-    /// Global retry logic used for all the backends
+    /// Retry configuration for IHttpRetryPolicy that uses RetryAfter header when present.
     /// </summary>
-    public IRetryMechanism RetryMechanism { get => this._retryMechanism; }
+    public sealed class HttpRetryConfig
```

Comment:
I feel like this class should exist separately from the config. There will be users who will use the TextCompletion and TextEmbeddings standalone. As long as those are public, I think the retry config should be readily available as well.

Reply:
Yeah I agree, we were going back and forth on this, but I think that makes sense. cc @dluc